src/full/Agda/TypeChecking/Generalize.hs

```diff
@@ -357,15 +357,18 @@ pruneUnsolvedMetas genRecName genRecCon genTel genRecFields interactionPoints is
     prepruneError :: MetaId -> String -> TCM a
     prepruneError x code = do
       r <- getMetaRange x
-      let msg = "Congratulations! You have found an easter egg (#" ++ code ++ "). " ++
-                "Be the first to submit a self-contained test case (max 50 lines of code) " ++
-                "producing this error message to https://github.com/agda/agda/issues/3672 " ++
-                "to receive a small prize."
-          cause = "The error is caused by complicated dependencies between unsolved " ++
-                  "metavariables and generalized variables. In particular, this meta:"
+      let msg = case code of
+            "RFCX" -> "Failed to generalize because some of the generalized variables depend on an " ++
```

Comment:
Shouldn't we have a enum type instead of these string constants?

Reply:
The strings are indeed a bit icky. If Haskell had local datatypes, an enum type would be nice. Added three separate functions instead.